Reno 3903/update social media links component

isastettler opened the pull request:
Fixes [JIRA ticket 3903](https://newyorkpubliclibrary.atlassian.net/browse/RENO-3903)

## This PR does the following:

- Removes `getLinksData` function
- Removes SocialMediaLinksUtils.ts
- Updates `SocialMediaLinkDataProps` types
- Updates storybook
- Updates tests

**NOTE: I added some commented out code to the `storybook/main.ts` file. The code snipped would allow for storybook to read and automatically apply prop types declared as Typescript `type`. Those are needed for conditional types which now need manual type adding (see MultiSelect and SocialMediaLink). It is a little out of the scope of this work to include it and has not been extensively tested.

## How has this been tested?

 Tested on Storybook and unit tests.

## Accessibility concerns or updates

<!--- Describe any accessibility concerns or updates that were made that should be known. -->

-

### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have updated the Storybook documentation accordingly.
- [N/A] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

<!--- This step is done AFTER creating a PR. -->
<!--- Vercel creates a static Storybook preview URL once the PR is created. -->
<!--- That preview URL is added by Vercel as a comment. -->

- [ ] Review the [Vercel preview ](https://nypl-design-system-git-reno-3903-update-social-medi-9886be-nypl.vercel.app/?path=/docs/components-navigation-socialmedialinks--docs)

isastettler committed:
remove default data, make labelText and url required for linkData items

isastettler committed:
update SocialMedia types, storybook and tests

isastettler committed:
suggest global fix to catch ts type declarations in storybook

isastettler committed:
update IconNames type for SocialMediaLinkDataProp

isastettler committed:
update socialMediaLinks theme file

EdwinGuzman commented on src/components/SocialMediaLinks/SocialMediaLinks.stories.tsx:
```diff
@@ -124,28 +90,62 @@ export const WithControls: Story = {
-        <SocialMediaLinks {...args} />
+  parameters: {
+    controls: {
+      hideNoControlsWarning: true,
```
Nice, we should consider doing this in the global config because I can see this being helpful in all component stories.

EdwinGuzman commented on src/components/SocialMediaLinks/SocialMediaLinks.stories.tsx:
```diff
@@ -11,65 +12,31 @@ import Heading from "../Heading/Heading";
 import Text from "../Text/Text";
 import { Source } from "@storybook/blocks";
 
+const exampleLinkData: SocialMediaLinkDataProps[] = [
```
This is great but all examples use this array data. Create two more: one has a different set and number of social media links, and the other has the "full" set of social media links to show more use cases.

isastettler replied:
Added.

EdwinGuzman commented on src/components/SocialMediaLinks/SocialMediaLinks.tsx:
```diff
@@ -15,24 +14,33 @@ export type ColorType = typeof colorTypeArray[number];
+
+// Verify that SocialMediaIconNames is a subset of IconNames
+type VerifyIconNames = IconNames extends DsIconNames ? true : false;
+// TYPE SAFETY: verified will throw an error if a value has been added to socialMediaIconMap that is not a IconName
```
There's no reference to `socialMediaIconMap`. Did you mean the local `IconNames` type?

isastettler replied:
Yup thank you - fixed.

EdwinGuzman commented on src/components/SocialMediaLinks/SocialMediaLinks.tsx:
```diff
@@ -15,24 +14,33 @@ export type ColorType = typeof colorTypeArray[number];
-export interface SocialMediaLinksProps {
-  /** Optional border: straight, circular or none. */
-  borders?: BorderType;
+interface CommonSocialMediaLinksProps {
```
nit: our convention is to call this "base" instead of "common"

isastettler replied:
Changed.

EdwinGuzman commented on src/components/SocialMediaLinks/SocialMediaLinks.tsx:
```diff
@@ -42,52 +50,23 @@ export interface SocialMediaLinksProps {
  * The SocialMediaLinks component renders a list of links for accessing social media sites.
+ * The data passed to the `linksData` prop controls which social media types are displayed,
+ * the order they appear, the url they link to, and the text used for the label of each.
+ * Order is determined by the order of the array.
```
You can remove the last sentence, it's covered in the previous sentence.

isastettler replied:
Done.

EdwinGuzman reviewed (commented):
Very minor updates but otherwise, the update looks good.

isastettler committed:
Merge branch 'development' into RENO-3903/update-social-media-links-component

isastettler committed:
update code as requested in peer review

isastettler commented:
As per Slack discussed I removed the `Source` components in the .stories.tsx files. To have a code insight available I removed the `sourceState={"none"}` that was added in the .mdx file to the `Canvas` components to remove the default `show code` buttons. I am adding this for visibility - if it does not seem to work I can revert it.

EdwinGuzman commented on .storybook/main.ts:
```diff
@@ -59,9 +59,24 @@ const config: StorybookConfig = {
-        // This is a bit of a "hack" but it tells Storybook that the prop
-        // is defined in a DS component. If we define it, then display it
-        // in the `ArgTypes` table in a component Storybook document page.
+        /** @Info: The following code would allow for typescript `type`
```
This is out of scope for this PR but I want you to keep it here commented out. We can make a new ticket to uncomment it and test it on all components so it can be tested separately from the `SocialMediaLinks` component. I would prefer that since this configuration affects all components. Again, leave it here to address later.

isastettler replied:
Yes that's what I thought too. Would be cool if it worked but needs definitely more testing.

EdwinGuzman reviewed (approved):
Thanks! Good updates.

isastettler committed:
Merge branch 'development' into RENO-3903/update-social-media-links-component

bigfishdesign13 commented on src/components/SocialMediaLinks/SocialMediaLinks.tsx:
```diff
@@ -15,24 +14,33 @@ export type ColorType = typeof colorTypeArray[number];
-  type: SocialMediaLinkType;
-  /** Optional override for default social media url */
-  url?: string;
+  type: IconNames;
```
I feel we are mixing terminology here and `type` as a prop name is no longer accurate.  If we are going to use the values from the `Icon` component's `name` prop, I would prefer to change this prop name to `icon`.   Thoughts?

EdwinGuzman replied:
Yes, that makes sense to me.

isastettler replied:
Updated the prop name and adjusted storybook and tests.

bigfishdesign13 reviewed (commented):
The functionality is great.  I just have a question about one of the prop names.

isastettler committed:
raplace "type" prop on linkData with "icon", update storybook and tests

bigfishdesign13 commented on src/components/SocialMediaLinks/SocialMediaLinks.tsx:
```diff
@@ -34,8 +34,8 @@ const verified: VerifyIconNames = true;
   labelText: string;
-  /** Required. Must be one of socialMediaLinkTypeArray */
-  type: IconNames;
+  /** Required. Should match the social media paltform that it's linking to */
```
💯 Love this.

isastettler replied:
Ugh. Fixed the typo. Thank you.

bigfishdesign13 replied:
Ha!  I didn't even notice the typo. 👀

bigfishdesign13 reviewed (approved):
Awesome.

isastettler committed:
fix typo

isastettler committed:
Merge branch 'development' into RENO-3903/update-social-media-links-component

bigfishdesign13 committed:
Merge branch 'development' into RENO-3903/update-social-media-links-component

bigfishdesign13 committed:
Merge branch 'development' into RENO-3903/update-social-media-links-component
